### PR TITLE
This branch implements a required parameter called 'access_token' for all API requests.

### PR DESCRIPTION
**Pull request by nickwxo**
This branch implements a required parameter called 'access_token' for all API requests. The token is then checked in a new middleware to find the relevant role for the application and the ACL is applied accordingly. If the token is not supplied or is invalid - an appropriate error message is returned instead. This branch also adds a role field to the App model with the same default role that matjames007 implemented.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -47,6 +53,27 @@ var app_manager = require('./routes/app/router-app-manager'),
     user = require('./routes/user/router-user'),
     farmer = require('./routes/resources/farmer/router-farmer');
 
+var roleMiddleware = function (req, res, next) {//this function gets the role name of the application whose access token was used in the API request
```
Comment (matjames007) [inline app.js]: @nick22891 We have to do some work on this one.  Token Authorization must be handled using another passport framework - and not a custom written middleware.  Checkout [this](https://www.npmjs.com/package/passport-token-auth).

File: public/javascript/controllers/user-ctrl.js [hunk 2]
```diff
@@ -90,12 +90,19 @@ angular.module('harvestv2')
             });
 
             $scope.createApp = function () {
+                PlatformFactory.show(function(info) {
+                    console.log(info);
```
Comment (matjames007) [inline public/javascript/controllers/user-ctrl.js]: @nick22891 remove console logs

**Comment by nickwxo** [comment]
Hey @matjames007 I did some reading and you can review this as-is. If the user sets 'authorization' in the header to the access token value the library uses it.

File: app.js [hunk 3]
```diff
@@ -47,6 +53,13 @@ var app_manager = require('./routes/app/router-app-manager'),
     user = require('./routes/user/router-user'),
     farmer = require('./routes/resources/farmer/router-farmer');
 
+
```
Comment (matjames007) [inline app.js:56]: @nick22891 please verify my changes.  Also work harder on commenting your code.

Comment (nickwxo) [reply app.js:56]: @matjames007 I've verified them. This branch still works as intended. Point taken about the code comments in future.

**Pull request merged**